feature-libs/asm/components/customer-emulation/customer-emulation.component.ts

```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
     );
     this.isCustomerEmulationSessionInProgress$ =
       this.asmComponentService.isCustomerEmulationSessionInProgress();
+
+    this.cartId.valueChanges.subscribe((response: string) => {
+      this.cartIdExists = response.trim().length > 0;
+    });
+
+    this.activeCartFacade.getActiveCartId().subscribe((response) => {
+      if (response) {
+        this.cartId.setValue(response);
+      }
+    });
   }
 
   logoutCustomer() {
     this.asmComponentService.logoutCustomer();
   }
 
+  /**
+   * Assign the input cart number to the customer
+   */
+  assignCartToCustomer() {
+    const customerId = this.customer.uid;
+    const cartId = this.cartId.value;
+
+    if (customerId) {
+      this.occAsmAdapter.bindCart(cartId, customerId).subscribe(
+        () => {
+          this.showAssignCartSuccess = true;
```

Comment:
There is a standard way of displaying success and error messages.  It would be more consistent to use them.  Let's discuss.

Reply:
Discuss with UX on using existing Spartacus toast messages